buf/registry/policy/v1beta1/configuration.proto

```diff
@@ -0,0 +1,144 @@
+// Copyright 2023-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.registry.policy.v1beta1;
+
+import "buf/validate/validate.proto";
+
+option go_package = "buf.build/gen/go/bufbuild/registry/protocolbuffers/go/buf/registry/policy/v1beta1";
+
+// The Configuration of a specific Policy at a specific Version. This consists of the built-in
+// rules version and lint, breaking, and plugin configurations required to run the Policy.
+message Configuration {
+  // The built-in rules version. This specifies which version of built-in rules to use for the Policy.
+  enum BuiltInVersion {
+    BUILT_IN_VERSION_UNSPECIFIED = 0;
+    BUILT_IN_VERSION_V1BETA1 = 1;
+    BUILT_IN_VERSION_V1 = 2;
+    BUILT_IN_VERSION_V2 = 3;
+  }
```

Comment:
This allows the user to specify the lint/breaking rules version they are using, which is tied to their `buf.yaml` configuration version currently. Should we allow this or should all policies default to `v2` built-in rule sets?

Reply:
Resolved this in our last discussion, we will be defaulting to the `v2` built-in rule set for policy checks. We are also uploading the entire config file instead of a proto message.